include/pybind11/pytypes.h

```diff
@@ -491,7 +491,7 @@ class tuple : public object {
         if (!m_ptr) pybind11_fail("Could not allocate tuple object!");
     }
     size_t size() const { return (size_t) PyTuple_Size(m_ptr); }
-    detail::tuple_accessor operator[](size_t index) const { return detail::tuple_accessor(ptr(), index); }
```

Comment:
Ok, I committed a few more cleanups that replace ``ptr()`` by ``this`` -- let me know if there are still some more that bother you and which I haven't caught.